src/main/java/org/opentripplanner/middleware/controllers/api/ApiController.java

```diff
@@ -30,57 +29,106 @@
  *  that exist in other collection).
  * @param <T> One of the {@link Model} classes (extracted from {@link TypedPersistence})
  */
-public abstract class ApiController<T extends Model> {
+public abstract class ApiController<T extends Model> implements Endpoint {
     private static final String ID_PARAM = "/:id";
     private final String ROOT_ROUTE;
     private static final String SECURE = "secure/";
     private static final Logger LOG = LoggerFactory.getLogger(ApiController.class);
     private final String classToLowercase;
-    private final TypedPersistence persistence;
+    private final TypedPersistence<T> persistence;
     private final Class<T> clazz;
 
     /**
      * @param apiPrefix string prefix to use in determining the resource location
      * @param persistence {@link TypedPersistence} persistence for the entity to set up CRUD endpoints for
      */
-    public ApiController (String apiPrefix, TypedPersistence persistence) {
+    public ApiController (String apiPrefix, TypedPersistence<T> persistence) {
         this.clazz = persistence.clazz;
         this.persistence = persistence;
         this.classToLowercase = persistence.clazz.getSimpleName().toLowerCase();
         this.ROOT_ROUTE = apiPrefix + SECURE + classToLowercase;
-        registerRoutes();
     }
 
     /**
-     * Register basic CRUD HTTP endpoints for the controller implementation.
+     * This method is called by spark-swagger to register endpoints and generate the docs.
+     * @param restApi The object to which to attach the documentation.
      */
-    private void registerRoutes() {
-        LOG.info("Registering routes for {}", ROOT_ROUTE);
-        // Options response for CORS
-        options(ROOT_ROUTE, (q, s) -> "");
-        // Get multiple entities.
-        get(ROOT_ROUTE, this::getMany, JsonUtils::toJson);
-        // Get one entity.
-        get(ROOT_ROUTE + ID_PARAM, this::getOne, JsonUtils::toJson);
-        // Create entity request
-        post(ROOT_ROUTE, this::createOrUpdate, JsonUtils::toJson);
-        // Update entity request
-        put(ROOT_ROUTE + ID_PARAM, this::createOrUpdate, JsonUtils::toJson);
-        // Delete entity request
-        delete(ROOT_ROUTE + ID_PARAM, this::deleteOne, JsonUtils::toJson);
+    @Override
+    public void bind(final SparkSwagger restApi) {
+        LOG.info("Registering routes and enabling docs for {}", ROOT_ROUTE);
+
+        restApi.endpoint(endpointPath(ROOT_ROUTE)
+                .withDescription("Interface for querying and managing '" + classToLowercase + "' entities."), (q, a) -> LOG.info("Received request for '{}' Rest API", classToLowercase))
```

Comment:
I think that log statement only gets called on a get all users call. Is that intentional?

Reply:
The log statement is called when swagger initializes the end points.